### DISCUSSION TITLE
使用MosDNS+AdGuardHome+MihomoTProxy实现DNS去广告与DNS分流

### DISCUSSION THREAD

**Discussion by jklolixxs** [nikkinikki-org/OpenWrt-nikki#197] [Show and tell]
MosDNS设置：最重要的就是 **_取消勾选DNS 转发_**
![firefox_WNENngMDl1](https://github.com/user-attachments/assets/ab8dace7-e8cf-4898-969c-4fc919817e27)
高级设置如图所示
![firefox_KgZnykwEIM](https://github.com/user-attachments/assets/f0da3db2-a1e1-478d-8318-059a64171695)
其余设置请自行更改

AdGuardHome：最重要的就是 **_重定向选项选择无_** 
![firefox_85TL5cR8vj](https://github.com/user-attachments/assets/b2c632fb-c385-4c63-8994-f10ff716cd73)
AdGuardHome内设置如图所示
![firefox_DozDnMqnxR](https://github.com/user-attachments/assets/96506a19-4f4a-4ee9-b27c-dcf79cc81763)
![firefox_IbFtgX3QLb](https://github.com/user-attachments/assets/5cbfa4be-d49c-43ef-b2cf-ceadf39a01d7)
其余设置，例如屏蔽广告等，请自行更改

MihomoTProxy：最重要的就是 **_勾选DNS劫持_** ，
![firefox_MmWjKDDYUL](https://github.com/user-attachments/assets/34d507c0-fd94-4841-ab9b-142972db0c8f)
![firefox_bO9fPEENtW](https://github.com/user-attachments/assets/abc65bfb-233b-48dd-a64d-7b52c1f4d9fa)
![firefox_uzd2aQuIqH](https://github.com/user-attachments/assets/4195661a-1f86-4f3a-9530-ebe1cc48c68f)
其余设置请自行更改

至此，三者已经相互关联起来了，全部启动即可

**Comment by Stefanleedan** [comment]
想请问下大佬，如果只单独使用adguard的DNS分流功能，不使用adg的去广告功能，也不使用Mosdns，请问该如何设置才能让Mihomo和adguard联动？

**Reply by jklolixxs** [reply]
> 想请问下大佬，如果只单独使用adguard的DNS分流功能，不使用adg的去广告功能，也不使用Mosdns，请问该如何设置才能让Mihomo和adguard联动？

adguardhome吗？还是手机或者电脑的adguard？

**Reply by Stefanleedan** [reply]
就是软路由里的adguardhome，之前用openclash，搭配adguardhome作为dns缓存用，用的挺好。现在换成mihomo，却不知道如何设置了。我也不用mosdns分流之类的插件，就只用adg，请问现在mihomo搭配adg，该如何配置呢？

**Reply by jklolixxs** [reply]
> 就是软路由里的adguardhome，之前用openclash，搭配adguardhome作为dns缓存用，用的挺好。现在换成mihomo，却不知道如何设置了。我也不用mosdns分流之类的插件，就只用adg，请问现在mihomo搭配adg，该如何配置呢？

你之前只用adguardhome？并且adguardhome执行DNS分流？这做不到吧，adgh没有这个功能啊。如果只用adgh和mihomo，那我只会mihomo分流，adgh做DNS缓存器

**Comment by lcese** [comment]
使用此方案完美实现一直想要实现的效果。

**Comment by refined-fish** [comment]
去广告与分流mihomo本身也可以完成，有没有朋友介绍一下这种模式的优点呀

**Reply by Borsalinolll** [reply]
其实不开启大陆绕过核心的模式下，这些工作mihomo全能独自完成，套娃只是多了个dns真实ip缓存，去广告效果mihomo加个规则集reject掉更好，adghome去广告纯副作用 费电没效果

**Comment by aihdde** [comment]
这样使用，mihomo里的遵循节点起不了作用啊

**Comment by renjiangsheng** [comment]
完全按照你的步骤操作，现在可以科学，ADGH 也有查询记录。好像国内网站全部不能访问，是哪里要改下设置吗？

**Reply by renjiangsheng** [reply]
并且设置后在Mihomo -dns设置-编辑 Fake-IP 过滤列表，添加geosite:private 或geosite:cn都会导致mihomo 无法运行。

**Comment by waxingren** [comment]
层层套娃了。再来个内网穿透   人都懵了   先试试

**Comment by aihdde** [comment]
你这种方式会导致Adguardhome里的拦截记录全是127.0.0.1
正确的方式是Adguardhome设置重定向53端口到adguardhome,nikki里正常设置国内国外dns,adguardhome上游dns设置为127.0.0.1:1053,再加一个备用的公共dns,防止nikki挂掉上不了国内,如此之后adguardhome里的拦截记录也可以分类出各个客户端

**Reply by aihddelyy** [reply]
DHCP可以由openwrt或ikuai接管,ADGH不提供dhcp

**Reply by nien0202** [reply]
> 你这种方式会导致Adguardhome里的拦截记录全是127.0.0.1 正确的方式是Adguardhome设置重定向53端口到adguardhome，nikki里正常设置国内国外dns，adguardhome上游dns设置为127.0.0.1：1053，再加一个备用的公共dns，防止nikki挂掉上不了国内，如此之后adguardhome里的拦截记录也可以分类出各个客户端

adguardhome上游dns设置为127.0.0.1:1053，但是测试上游服务器不可用，请问这是什么情况？openwrt恢复了出厂设置，简单设置了wan口可以拨号上网，别的没动，也还是不行。

**Reply by hnsycxm** [reply]
能整个详细的设置贴图吗

**Comment by 625045508** [comment]
你 nikki 的 IPv4  DNS劫持勾选了， AdguardHome 就无法接管内网设备DNS请求了

**Comment by howayneho** [comment]
Nikki先做了劫持，然后指向adg，adg指向了mos，mos的dns解析找谁呢？
分流nikki做了，adg拦截广告，mos干嘛用呢

**Comment by Michaelwuzb** [comment]
有没有办法出一期， mosdns和nikki直接配合的，不需要adg那东西感觉没啥用。

**Reply by joyoner** [reply]
同需要

**Comment by basaltfiber** [comment]
为啥nikki要开启dns劫持？
